int8 type
int64 spec
int8 inUse
